Watcher crash: invalid memory address or nil pointer dereference

GitHub checks not getting updated because `watcher` crashes with the following dump:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x18c0b44]

goroutine 131 [running]:
github.com/openshift-pipelines/pipelines-as-code/pkg/pipelineascode.SecretFromRepository({0x2258758, 0xc00069cc90}, 0xc0000a9040, {0x2258950, 0xc00054e778}, 0x0, 0xc0014c6c60, 0xc00173e580, 0xc00069cc90)
	/go/src/github.com/openshift-pipelines/pipelines-as-code/pkg/pipelineascode/secret.go:32 +0x104
github.com/openshift-pipelines/pipelines-as-code/pkg/reconciler.(*Reconciler).reportStatus(0xc0000f9530, {0x2258758, 0xc00069cc90}, 0xc000133530, 0xc0014c6c60, 0xc001301400, {0x2298a00, 0xc0014bf7c0})
	/go/src/github.com/openshift-pipelines/pipelines-as-code/pkg/reconciler/reconciler.go:71 +0x1e5
github.com/openshift-pipelines/pipelines-as-code/pkg/reconciler.(*Reconciler).ReconcileKind(0xc0014bf700, {0x2258758, 0xc00069cc90}, 0xc001301400)
	/go/src/github.com/openshift-pipelines/pipelines-as-code/pkg/reconciler/reconciler.go:57 +0x468
github.com/tektoncd/pipeline/pkg/client/injection/reconciler/pipeline/v1beta1/pipelinerun.(*reconcilerImpl).Reconcile(0xc0005f9360, {0x2258758, 0xc00069cc60}, {0xc001430030, 0x22})
	/go/src/github.com/openshift-pipelines/pipelines-as-code/vendor/github.com/tektoncd/pipeline/pkg/client/injection/reconciler/pipeline/v1beta1/pipelinerun/reconciler.go:235 +0x6a2
knative.dev/pkg/controller.(*Impl).processNextWorkItem(0xc000135500)
	/go/src/github.com/openshift-pipelines/pipelines-as-code/vendor/knative.dev/pkg/controller/controller.go:542 +0x5a5
knative.dev/pkg/controller.(*Impl).RunContext.func3()
	/go/src/github.com/openshift-pipelines/pipelines-as-code/vendor/knative.dev/pkg/controller/controller.go:478 +0x68
created by knative.dev/pkg/controller.(*Impl).RunContext
	/go/src/github.com/openshift-pipelines/pipelines-as-code/vendor/knative.dev/pkg/controller/controller.go:476 +0x2de
```

This was working as expected but I don't have a date/version when this stopped working. Repository has the minimal configuration (url).

We are running OCP 4.11 with Pipeline Operator v1.8

--- a/pkg/pipelineascode/match.go
+++ b/pkg/pipelineascode/match.go
@@ -59,13 +59,13 @@ func (p *PacRun) matchRepoPR(ctx context.Context) ([]matcher.Match, *v1alpha1.Re
 	// allow webhook providers users to have a global webhook secret to be used,
 	// so instead of having to specify their in Repo each time, they use a
 	// shared one from pac.
-	if repo.Spec.GitProvider != nil && p.event.InstallationID <= 0 {
+	if p.event.InstallationID > 0 {
+		p.event.Provider.WebhookSecret, _ = GetCurrentNSWebhookSecret(ctx, p.k8int)
+	} else {
 		err := SecretFromRepository(ctx, p.run, p.k8int, p.vcx.GetConfig(), p.event, repo, p.logger)
 		if err != nil {
 			return nil, nil, err
 		}
-	} else {
-		p.event.Provider.WebhookSecret, _ = GetCurrentNSWebhookSecret(ctx, p.k8int)
 	}
 
 	// validate payload  for webhook secret

--- a/pkg/pipelineascode/secret.go
+++ b/pkg/pipelineascode/secret.go
@@ -23,12 +23,18 @@ const (
 // SecretFromRepository grab the secret from the repository CRD
 func SecretFromRepository(ctx context.Context, cs *params.Run, k8int kubeinteraction.Interface, config *info.ProviderConfig, event *info.Event, repo *apipac.Repository, logger *zap.SugaredLogger) error {
 	var err error
+	if repo.Spec.GitProvider == nil {
+		return fmt.Errorf("failed to find git_provider details in repository spec: %v/%v", repo.Namespace, repo.Name)
+	}
 	if repo.Spec.GitProvider.URL == "" {
 		repo.Spec.GitProvider.URL = config.APIURL
 	} else {
 		event.Provider.URL = repo.Spec.GitProvider.URL
 	}
 
+	if repo.Spec.GitProvider.Secret == nil {
+		return fmt.Errorf("failed to find secret in git_provider section in repository spec: %v/%v", repo.Namespace, repo.Name)
+	}
 	gitProviderSecretKey := repo.Spec.GitProvider.Secret.Key
 	if gitProviderSecretKey == "" {
 		gitProviderSecretKey = defaultGitProviderSecretKey

